### DISCUSSION TITLE
a-deep-dive-on-stringbuilder-part-5-reducing-allocations-by-caching-stringbuilders-with-stringbuildercache/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#28] [Announcements]
# a-deep-dive-on-stringbuilder-part-5-reducing-allocations-by-caching-stringbuilders-with-stringbuildercache/

In this post I look at the internal StringBuilderCache type, benchmark it against a StringBuilder, and walk through the code to show how it works.

https://andrewlock.net/a-deep-dive-on-stringbuilder-part-5-reducing-allocations-by-caching-stringbuilders-with-stringbuildercache/

**Comment by DanielKeogh** [comment]
The reason that Microsoft doesn't publish StringBuilderCache is probably the same reason that I had to bin my own StringBuilderCache a few years back: Turns out that clearing and re-using a StringBuilder can have massive performance problems if you do anything other than just append to the end of it.

Certain inserts or delete in the middle of a string builder makes it freak out when you call clear, making it slower to clear than just to create a new string builder. (Or at least that was what happened in my production system, forcing me to bin the cache).

[a comment by an automated account is not shown]
